samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <nrf_modem_gnss.h>
+#include <event_manager.h>
+
+#include "gnss_module.h"
+#include "gnss_pvt_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(gnss_module, CONFIG_APP_LOG_LEVEL);
+
+static struct nrf_modem_gnss_pvt_data_frame pvt_data;
+
+/**@brief Callback for GNSS events */
+static void gnss_event_handler(int event_id)
+{
+	int err = 0;
+
+	switch (event_id) {
+	case NRF_MODEM_GNSS_EVT_PVT: {
+		err = nrf_modem_gnss_read(&pvt_data, sizeof(pvt_data), NRF_MODEM_GNSS_DATA_PVT);
```

Comment:
This code could be moved to case `NRF_MODEM_GNSS_EVT_FIX` instead. It gets the same PVT data, so you could just remove the `pvt_data.flags & NRF_MODEM_GNSS_PVT_FLAG_FIX_VALID` check.